price_parser/parser.py

```diff
@@ -189,6 +193,34 @@ def extract_price_text(price: str) -> Optional[str]:
     >>> extract_price_text("50")
     '50'
     """
+    price_amount = _extract_price_amount(price, currency)
+    return price_amount.text
+
+
+def _extract_price_amount(price: str, currency: Optional[str] = "") -> Optional[str]:
+    """
+    Extract from a string the text of a price and a flag indicating
+    if this is a string of a negative price.
+    """
+
+    PriceAmount = namedtuple('PriceAmount', ['text', 'negative'])
+
+    if price is None:
+        return PriceAmount(text=None, negative=False)
+
+    negative_regexes = [
+        r"-\s*?\d[\d.,\d]*",
+        r"\d[\d.,\d]*\d-",
+    ]
+    if currency is not None:
+        negative_regexes.append(r"-{}\d[\d.,\d]*".format(re.escape(currency)))
+    negative_amount_search = re.search(
+        r"({})(?:[^%\d]|$)".format("|".join(negative_regexes)),
+        price,
+        re.VERBOSE
+    )
+    negative_amount = bool(negative_amount_search)
```

Comment:
So we are determining if the amount is negative or positive before we have extracted the amount. I think this may be a problem, for example once https://github.com/scrapinghub/price-parser/pull/5 is merged, where `- 2 items at 24,00 €` could parse the price as `-24€` (I did not check, though).

Reply:
You broke my tests! :-( 
You are right. This scenario will fail doing that way. However I don't see how to identify it after extracting the amount. I will include it to the tests and try to solve it.